### PR TITLE
Feature/initial course docs

### PR DESCRIPTION
**Pull request by philratcliffe**
This is the initial implementation of the Course Dataset API.

Testing:
Run the Azure function and call it using both valid and invalid parameters.

You can run the tests by using the nose Python testing module. To setup the required environment follow the instructions in the tests.py module.

### CODE REVIEW

File: CoursesApiHttpTrigger/function.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "bindings": [
+    {
+      "authLevel": "function",
+      "type": "httpTrigger",
+      "direction": "in",
+      "name": "req",
+      "route": "institutions/{institution_id}/courses/{course_id}/modes/{mode}",
+      "methods": [
+        "get",
+        "post"
```
Comment (nshumoogum) [inline CoursesApiHttpTrigger/function.json]: Should just be a get request :)

File: CoursesApiHttpTrigger/tests.py [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+        # Do some checking of the returned error message.
+        error_msg = json.loads(resp.get_body().decode('utf-8'))
+        print(error_msg)
+        self.assertEqual(error_msg['errors'][0]["error"], "Bad Request")
+        self.assertEqual(
+            error_msg['errors'][0]["error_values"][0]['Parameter Error'],
+            'Invalid parameter passed')
+
+    # TODO add more tests
```
Comment (nshumoogum) [inline CoursesApiHttpTrigger/tests.py:409]: So for mode i would add test to fail when `mode` is set to a value of `one`

File: CoursesApiHttpTrigger/utils.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    # Get the relevant environment variables
+    cosmosdb_uri = os.environ['AzureCosmosDbUri']
+
+    # This should use the read only key (least privilege)
+    cosmosdb_key = os.environ['AzureCosmosDbKey']
+
+    master_key = 'masterKey'
+
+    # Initialise anf return the Cosmos client
```
Comment (nshumoogum) [inline CoursesApiHttpTrigger/utils.py]: Spelling mistake `anf` instead of `and`

**Review by nshumoogum — CHANGES_REQUESTED** [review]
Good work, only a couple of comments

File: CoursesApiHttpTrigger/function.json (hunk 1 above)
Comment (philratcliffe) [reply CoursesApiHttpTrigger/function.json]: Well spotted :-)

File: CoursesApiHttpTrigger/tests.py (hunk 2 above)
Comment (philratcliffe) [reply CoursesApiHttpTrigger/tests.py:409]: Done.

File: CoursesApiHttpTrigger/utils.py (hunk 3 above)
Comment (philratcliffe) [reply CoursesApiHttpTrigger/utils.py]: Fixed.

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**